yw_website/yw_website/apps/website/models.py

```diff
@@ -102,3 +102,73 @@ class TagFile(models.Model):
 
     class Meta:
         unique_together = ('tag', 'file')
+
+class ProgramBlock(models.Model):
+    class Meta:
+        unique_together = ('programblock_id', 'version')
+    programblock_id = models.IntegerField()
+    name = models.TextField()
+    qualified_name = models.TextField()
+    in_program_block = models.ForeignKey(
```

Comment:
this wont be an explicit foreign key restraint because of the jankiness we talked about yesterday

Reply:
I disagree I think it still can be as long as there is an order to the API, eg. parent program blocks always coming first in the JSON. Since we have guaranteed uniqueness on the progblock_id and version foreign key we can track these relationships

Reply:
Although I think we should use runs instead of version for these models

Reply:
no, that is incorrect.  probrambock_id is not gauranteed to be unique. the combination of programblock_id and version are gauranteed to be unique.

Reply:
> Although I think we should use runs instead of version for these models

why? we discussed that 'uri' information is unique per run, and non uri information, aka model data (program block, data, and channel) is unique per version.

Reply:
I think you're conflating this with being a raw integer id when it is instead a reference to a separate ORM object of probramBlock type

Reply:
Also since the database is built and torn down for each run in yw_save I think the only way to guarantee uniqueness is to make a unique composite key on a run foreign key over version

Reply:
will that mapping be apparent when you write the serializer?